### ISSUE TITLE
int size not always clear

### ISSUE THREAD

**Issue by janbrandejs** [TAPPorg/tensor-interfaces#14]
In some parts of the interface, int size is not specified, e.g. int nmode in tensor.h. Anthony Scemama recommends to speficy this everywhere in the interface.

**Comment by devinamatthews** [comment]
Any reason to prefer int32_t or int64_t for nmodes?

**Comment by janbrandejs** [comment]
no! also, i think we should never use int itself without a clear size. I'm using multiple languages, libs and machines and it would be better to have this clear everywhere

**Comment by devinamatthews** [comment]
Is int32_t OK though? Seems a bit odd to use int64_t for this but I guess it would be consistent.

**Comment by janbrandejs** [comment]
We can either 
1. use int64_t everywhere
2. or use int32_t for nmodes, int64_t for extents and strides, char for indices, some other sizes for pointers and error codes.

I'd do 1. because I when I was interfacing TAPP to fortran code I was tired from having to check this separately for every parameter, as there is many of them.

**Comment by evaleev** [comment]
I vote 1, i.e. 64-bit int for all integers.
